membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -62,25 +47,30 @@ object AccountDetails {
       }
 
       Json.obj(
-        "joinDate" -> paymentDetails.startDate,
-        "optIn" -> !paymentDetails.pendingCancellation,
-        "subscription" -> (paymentMethod ++ Json.obj(
-          "start" -> paymentDetails.lastPaymentDate,
-          "end" -> endDate,
-          "nextPaymentPrice" -> paymentDetails.nextPaymentPrice,
-          "nextPaymentDate" -> paymentDetails.nextPaymentDate,
-          "renewalDate" -> paymentDetails.termEndDate,
-          "cancelledAt" -> (paymentDetails.pendingAmendment || paymentDetails.pendingCancellation),
-          "subscriberId" -> paymentDetails.subscriberId,
-          "trialLength" -> paymentDetails.remainingTrialLength,
-          "plan" -> Json.obj(
-            "name" -> paymentDetails.plan.name,
-            "amount" -> paymentDetails.plan.price.amount * 100,
-            "currency" -> paymentDetails.plan.price.currency.glyph,
-            "currencyISO" -> paymentDetails.plan.price.currency.iso,
-            "interval" -> paymentDetails.plan.interval.mkString
-          )))
-      )
+        "tier" -> paymentDetails.plan.name,
+        "isPaidTier" -> (paymentDetails.plan.price.amount > 0f)
+      ) ++
+        regNumber.fold(Json.obj())({reg => Json.obj("regNumber" -> reg)}) ++
+        Json.obj(
+          "joinDate" -> paymentDetails.startDate,
+          "optIn" -> !paymentDetails.pendingCancellation,
```

Comment:
Do we need this confusing field if nothing in the manage. site uses it?

Reply:
there appear to be a few _possible_ usages https://github.com/search?q=org%3Aguardian+optIn&type=Code

however, when the deprecated endpoints come to be removed (following everything moving to `manage-frontend`) it will be wise to do a serious spring-clean of this whole serialiser (once theres just one thing relying on it) as there are MANY improvements that could be made (removing things, renaming things, restructuring etc.)

Reply:
agree, we should spring clean, especially if the endpoint is being moved to a different URL.  I wonder now, based on our chat last time, whether we should totally change to using a Wire model with a suitable hierarchy, rather than having sucha complicated serialiser (that additionaliy loses the account->subscription one to many relationship.)